### PR TITLE
switch from DecoratedImage to MaskedImage in makeFpSummary.py

### PR DESCRIPTION
**Pull request by jchiang87**

### CODE REVIEW

File: bin.src/makeFpSummary.py [hunk 1]
```diff
@@ -108,14 +108,13 @@ def callback(im, ccd, imageSource):
...
+                        binned_im = bi.getCcdImage(ccd, binSize=self.config.sensorBinSize,
+                                                   as_masked_image=True)[0]
                         binned_im = rotateImageBy90(binned_im, ccd.getOrientation().getNQuarter())
                         if self.config.putFullSensors:
-                            binned_dim = afwImage.DecoratedImageF(binned_im)
-                            if md:
-                                binned_dim.setMetadata(md)
-
-                            butler.put(binned_dim, 'binned_sensor_fits', **dataId, dstype=dstype)
+                            # binned_mim = afwImage.MaskedImageF(binned_im, metadata=md)
```
Comment (jchiang87) [inline bin.src/makeFpSummary.py]: The `metadata` kwd arg in the `MaskedImage` constructor wants to be a "PropertySet" but `md` here is a "PropertyList".  I couldn't figure how out to get the former from the latter, so I've omitted it in the uncommented code, which executes to completion, producing files with the desired mask planes.  I have no idea how the missing metadata is used downstream.

Comment (jchiang87) [reply bin.src/makeFpSummary.py]: When I use
```
binned_mim = afwImage.MaskedImageF(binned_im[box], metadata=md)
```
I get the following error output:
```
Traceback (most recent call last):
  File "/scratch/jchiang/DC2/fp_summary_fixes/obs_lsstCam/bin/makeFpSummary.py", line 180, in <module>
    FocalplaneSummaryTask.parseAndRun()
  File "/software/lsstsw/stack_20181012/stack/miniconda3-4.5.4-fcd27eb/Linux64/pipe_base/16.0-15-gb461e1a+7/python/lsst\
/pipe/base/cmdLineTask.py", line 603, in parseAndRun
    resultList = taskRunner.run(parsedCmd)
  File "/software/lsstsw/stack_20181012/stack/miniconda3-4.5.4-fcd27eb/Linux64/pipe_base/16.0-15-gb461e1a+7/python/lsst\
/pipe/base/cmdLineTask.py", line 221, in run
    resultList = list(mapFunc(self, targetList))
  File "/scratch/jchiang/DC2/fp_summary_fixes/obs_lsstCam/bin/makeFpSummary.py", line 36, in __call__
    result = task.runDataRef(dataRefList, butler=butler)
  File "/scratch/jchiang/DC2/fp_summary_fixes/obs_lsstCam/bin/makeFpSummary.py", line 128, in runDataRef
    binned_mim = afwImage.MaskedImageF(binned_im[box], metadata=md)
TypeError: __init__(): incompatible constructor arguments. The following argument types are supported:
    1. lsst.afw.image.maskedImage.maskedImage.MaskedImageF(width: int, height: int, planeDict: Dict[str, int]={})
    2. lsst.afw.image.maskedImage.maskedImage.MaskedImageF(dimensions: lsst.geom.Extent2I, planeDict: Dict[str, int]={}\
)
    3. lsst.afw.image.maskedImage.maskedImage.MaskedImageF(image: lsst.afw.image.image.image.ImageF, mask: lsst.afw.ima\
ge.image.image.MaskX=None, variance: lsst.afw.image.image.image.ImageF=None)
    4. lsst.afw.image.maskedImage.maskedImage.MaskedImageF(bbox: lsst.geom.Box2I, planeDict: Dict[str, int]={})
    5. lsst.afw.image.maskedImage.maskedImage.MaskedImageF(fileName: str, metadata: lsst.daf.base.propertyContainer.pro\
pertySet.PropertySet=None, bbox: lsst.geom.Box2I=Box2I(minimum=Point2I(0, 0), dimensions=Extent2I(0, 0)), origin: lsst.\
afw.image.image.image.ImageOrigin=ImageOrigin.PARENT, conformMasks: bool=False, needAllHdus: bool=False, imageMetadata:\
 lsst.daf.base.propertyContainer.propertySet.PropertySet=None, maskMetadata: lsst.daf.base.propertyContainer.propertySe\
t.PropertySet=None, varianceMetadata: lsst.daf.base.propertyContainer.propertySet.PropertySet=None, allowUnsafe: bool=F\
alse)
    6. lsst.afw.image.maskedImage.maskedImage.MaskedImageF(manager: lsst.afw.fits.fits.MemFileManager, metadata: lsst.d\
af.base.propertyContainer.propertySet.PropertySet=None, bbox: lsst.geom.Box2I=Box2I(minimum=Point2I(0, 0), dimensions=E\
xtent2I(0, 0)), origin: lsst.afw.image.image.image.ImageOrigin=ImageOrigin.PARENT, conformMasks: bool=False, needAllHdu\
s: bool=False, imageMetadata: lsst.daf.base.propertyContainer.propertySet.PropertySet=None, maskMetadata: lsst.daf.base\
.propertyContainer.propertySet.PropertySet=None, varianceMetadata: lsst.daf.base.propertyContainer.propertySet.Property\
Set=None, allowUnsafe: bool=False)
    7. lsst.afw.image.maskedImage.maskedImage.MaskedImageF(rhs: lsst.afw.image.maskedImage.maskedImage.MaskedImageF, de\
ep: bool=False)
    8. lsst.afw.image.maskedImage.maskedImage.MaskedImageF(rhs: lsst.afw.image.maskedImage.maskedImage.MaskedImageF, bb\
ox: lsst.geom.Box2I, origin: lsst.afw.image.image.image.ImageOrigin=ImageOrigin.PARENT, deep: bool=False)
    9. lsst.afw.image.maskedImage.maskedImage.MaskedImageF(src: lsst.afw.image.maskedImage.maskedImage.MaskedImageI, de\
ep: bool)
    10. lsst.afw.image.maskedImage.maskedImage.MaskedImageF(src: lsst.afw.image.maskedImage.maskedImage.MaskedImageD, d\
eep: bool)
    11. lsst.afw.image.maskedImage.maskedImage.MaskedImageF(src: lsst.afw.image.maskedImage.maskedImage.MaskedImageU, d\
eep: bool)
    12. lsst.afw.image.maskedImage.maskedImage.MaskedImageF(src: lsst.afw.image.maskedImage.maskedImage.MaskedImageL, d\
eep: bool)

Invoked with: lsst.afw.image.maskedImage.maskedImage.MaskedImageF=(image=[[ 5.43345    -2.0875816   1.2639545  ...  0.5\
219909  -0.67320067
  -7.87033   ]
 [ 8.047948    2.6326475   4.159735   ... -4.59726     6.4118567
  -0.7404918 ]
 [ 2.6665483   1.2349975  -0.98744977 ...  5.2068834   1.3243444
  -1.123787  ]
 ...
 [ 2.9466734  -6.8578773   7.9186277  ... -4.2724433  -7.5339394
  -7.1324377 ]
 [ 0.01870394 -1.7832227   2.993019   ...  7.6577444  -3.017055
  -0.7017913 ]
 [13.407105    6.357986    2.2930014  ...  2.7609234   8.950816
  -3.236945  ]],
mask=[[528 528 512 ... 512 528 528]
 [528 528 512 ... 512 528 528]
 [528 528 512 ... 512 528 528]
 ...
 [528 528 512 ... 512 528 528]
 [528 528 528 ... 528 528 528]
 [528 528 528 ... 528 528 528]], maskPlaneDict={'BAD': 0, 'CR': 3, 'CROSSTALK': 9, 'DETECTED': 5, 'DETECTED_NEGATIVE': \
6, 'EDGE': 4, 'INTRP': 2, 'NO_DATA': 8, 'SAT': 1, 'SUSPECT': 7, 'UNMASKEDNAN': 10}
variance=[[28.058342 27.789772 27.930515 ... 27.894781 27.977581 27.784376]
 [28.157555 27.972012 28.014744 ... 27.797543 28.30048  28.067028]
 [27.962143 27.895718 27.805206 ... 28.216232 28.13125  28.074074]
 ...
 [28.000475 27.631437 28.180025 ... 27.98402  27.871529 27.80255 ]
 [27.872269 27.79062  27.955038 ... 28.420643 27.9522   27.921507]
 [28.296227 28.058598 27.87155  ... 28.156656 28.253952 27.734468]], bbox=(minimum=(0, 500), maximum=(1017, 999))); kwa\
rgs: metadata=<lsst.daf.base.propertyContainer.propertyList.PropertyList object at 0x7f4d65a063b0>
```

Comment (TallJimbo) [reply bin.src/makeFpSummary.py]: Ah, I understand now - the `metadata` argument isn't allowed at all when the other argument is a `MaskedImage` (it's only relevant when you pass in a filename to read from disk).  When you pass in another `MaskedImage`, the only other argument you can provide is `deep`, to set whether to do a deep copy or not.

Comment (jchiang87) [reply bin.src/makeFpSummary.py]: Thanks for the clarification (I should have read the error message more carefully.)  After discussing with Simon, it's clear that we want to replace the `DecoratedImage` with an `Exposure`.  After doing so, everything seems to work fine.

**Comment by jchiang87** [comment]
@SimonKrughoff Can you review this please?  I can't seem to select you (or anyone else) as a reviewer.

File: bin.src/makeFpSummary.py [hunk 2]
```diff
@@ -126,10 +125,9 @@ def callback(im, ccd, imageSource):
                              'B': afwGeom.Box2I(afwGeom.PointI(0, 0), afwGeom.ExtentI(x, y/2))}
                     for half in ('A', 'B'):
                         box = boxes[half]
-                        binned_dim = afwImage.DecoratedImageF(binned_im[box])
-                        if md:
-                            binned_dim.setMetadata(md)
-                        butler.put(binned_dim, 'binned_sensor_fits_halves', half=half,
+                        # binned_mim = afwImage.MaskedImageF(binned_im[box], metadata=md)
```
Comment (SimonKrughoff) [inline bin.src/makeFpSummary.py]: Same here.

File: bin.src/makeFpSummary.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jchiang87) [reply bin.src/makeFpSummary.py]: These comments are on the outdated version of the code.  The current version already has the commented-out lines removed.

[the remark this answers is not in the document]
Comment (SimonKrughoff) [reply bin.src/makeFpSummary.py]: OK.  Sorry about that.

**Comment by jchiang87** [comment]
[DM-16654](https://jira.lsstcorp.org/browse/DM-16654) has been merged into master (thanks, Merlin), so this is ready to go.

**Comment by heather999** [comment]
Now rebased and ready to merge.

**Pull request merged**